sdk/devcenter/developer-devcenter-rest/test/public/devBoxTests.spec.ts

```diff
@@ -0,0 +1,541 @@
+      .get();
+
+    if (isUnexpected(poolOutput)) {
+      throw poolOutput.body.error;
+    }
+
+    expect(poolOutput.body.name).to.equal(poolName);
+  });
+
+  it("GetPools", async function () {
+    const poolsList = await client.path("/projects/{projectName}/pools", projectName).get();
+
+    if (isUnexpected(poolsList)) {
+      throw poolsList.body.error;
+    }
+
+    const pools: PoolOutput[] = [];
+    console.log("Iterating through pool results:");
+
+    for await (const pool of paginate(client, poolsList)) {
+      const { name } = pool;
+      console.log(`Received pool "${name}"`);
+      pools.push(pool);
+    }
+
+    expect(pools.length).to.equal(1);
+    expect(pools[0].name).to.equal(poolName);
+  });
+
+  it("GetSchedule", async function () {
+    await createDevBox();
+
+    const scheduleOutput = await client
+      .path(
+        "/projects/{projectName}/pools/{poolName}/schedules/{scheduleName}",
+        projectName,
+        poolName,
+        "default"
+      )
+      .get();
+
+    if (isUnexpected(scheduleOutput)) {
+      throw scheduleOutput.body.error;
+    }
+
+    expect(scheduleOutput.body.name).to.equal("default");
+
+    await deleteDevBox();
+  });
+
+  it("GetSchedules", async function () {
+    await createDevBox();
+
+    const schedulesListResponse = await client
+      .path("/projects/{projectName}/pools/{poolName}/schedules", projectName, poolName)
+      .get();
+
+    if (isUnexpected(schedulesListResponse)) {
+      throw schedulesListResponse.body.error;
+    }
+
+    const schedules: ScheduleOutput[] = [];
+    console.log("Iterating through schedules results:");
+
+    for await (const schedule of paginate(client, schedulesListResponse)) {
+      const { name } = schedule;
+      console.log(`Received schedule "${name}"`);
+      schedules.push(schedule);
+    }
+
+    expect(schedules.length).to.equal(1);
+    expect(schedules[0].name).to.equal("default");
+
+    await deleteDevBox();
+  });
+
+  it("GetDevBox", async function () {
+    await createDevBox();
+
+    const devboxOutput = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}",
+        projectName,
+        userId,
+        devboxName
+      )
+      .get();
+
+    if (isUnexpected(devboxOutput)) {
+      throw devboxOutput.body.error;
+    }
+
+    expect(devboxOutput.body.name).to.equal(devboxName);
+
+    await deleteDevBox();
+  });
+
+  it("GetRemoteConnection", async function () {
+    await createDevBox();
+
+    const remoteConnectionResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/remoteConnection",
+        projectName,
+        userId,
+        devboxName
+      )
+      .get();
+
+    if (isUnexpected(remoteConnectionResponse)) {
+      throw remoteConnectionResponse.body.error;
+    }
+
+    assert.isTrue(isAValidUrl(remoteConnectionResponse.body.rdpConnectionUrl as string));
```

Comment:
Would also recommend validating the `webUrl` you get back

Reply:
iirc the URL format for connection strings looks like this: https://{domain}/webclient/{activity-id}

I think it makes sense to validate whether or not the URL is valid. That we already do. Then also we can check if it contains "webclient" since everything else changes from dev box to dev-box.